RPGLC/com/rpglc/subevent/AttackRoll.cs

```diff
@@ -4,6 +4,63 @@
 
 namespace com.rpglc.subevent;
 
+/// <summary>
+///   Performs an attack roll against a target. Different results may occur on a hit or miss. Damage defined in "damage" is only dealt on a hit.
+///   
+///   <code>
+///   {
+///     "subevent": "attack_roll",
+///     "tags": [
+///       &lt;string&gt;
+///     ],
+///     "attack_type": &lt;string&gt;,
+///     "ability": &lt;string&gt;,
+///     "use_origin_ability": &lt;bool = false&gt;,
+///     "damage": [
+///       &lt;bonus formula&gt;
+///     ],
+///     "withhold_damage_modifier": &lt;bool = false&gt;,
+///     "vampirism": [
+///       &lt;vampirism formula&gt;
+///     ],
+///     "hit": [
+///       &lt;nested subevent&gt;
+///     ],
+///     "miss": [
+///       &lt;nested_subevent&gt;
+///     ]
+///   }
+///   </code>
+///   
+///   <list type="bullet">
+///     <item>"tags" is an optional field and will default to a value of [ ] if left unspecified. Any tags provided will be inherited by any nested subevents.</item>
```

Comment:
Indicate that "attack_type" is added to the tags array during subevent preparation.